docs/ref/checks.txt

```diff
@@ -141,6 +141,9 @@ Models
 * **models.E022**: ``<function>`` contains a lazy reference to
   ``<app label>>.<model>``, but app ``<app label>`` isn't installed or
   doesn't provide model ``<model>``.
+* **models.E023**: The model name ``<model>`` starts or ends with underscores.
+* **models.E024**: The model name ``<model>`` contians double underscores.
```

Comment:
Also, perhaps it is worth saying why this is not allowed, i.e. double underscores are the lookup / join separator.